packages/tupaia-web/src/Routes.tsx

```diff
@@ -7,23 +7,42 @@ import { Navigate, Route, Routes as RouterRoutes, useLocation } from 'react-rout
 import { ModalRoutes, LandingPage, Project } from './pages';
 import { MODAL_ROUTES, DEFAULT_URL } from './constants';
 import { Layout } from './layout';
+import { useUser } from './api/queries';
+import { LoadingScreen } from './components';
 
 /**
  * This Router is using [version 6.3]{@link https://reactrouter.com/en/v6.3.0}, as later versions are not supported by our TS setup. See [this issue here]{@link https://github.com/remix-run/react-router/discussions/8364}
  * This means the newer 'createBrowserRouter' and 'RouterProvider' can't be used here.
  *
- * **/
+ *
+ * * */
 
 export const Routes = () => {
-  let location = useLocation();
+  const location = useLocation();
+  const { isLoggedIn, isLoading } = useUser();
+  // if the user is loading, show a loading screen, so that we can handle redirects when this is done
+  if (isLoading)
```

Comment:
It's a bit of a shame that nothing really gets rendered if the user request is loading. While that is pretty unavoidable for the "/" route, it could be avoided for other routes. Could we move this is loading check into the redirect somehow. Maybe something like this

````
const HomeRedirect = () => {
  const { isLoading, isLoggedIn } = useUser();

  if (isLoading) {
    return <LoadingScreen isLoading />;
  }
  return (
    <Navigate
      to={`${DEFAULT_URL}#${isLoggedIn ? MODAL_ROUTES.PROJECTS : MODAL_ROUTES.LOGIN}`}
      replace
    />
  );
};

<Route path="/" element={<HomeRedirect />} />